src/Form/AccountForm.php

```diff
@@ -46,6 +82,45 @@ public function form(array $form, FormStateInterface $form_state) {
     return $form;
   }
 
+  /**
+   * Ajax callback to fetch the list of mpx accounts.
+   *
+   * @param array &$form
+   *   The form being rendered.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   The current state of the form.
+   *
+   * @return array
+   *   The accounts form item.
+   */
+  public function fetchAccounts(array &$form, FormStateInterface $form_state) : array {
+    $user_entity_id = $form_state->getValue('user');
+
+    /** @var \Drupal\media_mpx\Entity\UserInterface $user */
+    $user = $this->entityTypeManager->getStorage('media_mpx_user')->load($user_entity_id);
+
+    $accountFactory = $this->dataObjectFactory->forObjectType($user, 'Access Data Service', 'Account', '1.0');
```

Comment:
is this a factory method? If so, shouldn't this be from instead of for?

Reply:
Hm. Let's see.

> An account factory for the Access Data Service Account object, version 1.0

> An account factory from the Access Data Service Account object, version 1.0

I actually think I should rename `$this->dataObjectFactory` and possibly the class.. It's actually a _factory_ for a _factory_ (see the services.yml for how it works).

Reply:
Fine for me. Up to you.